### ISSUE TITLE
Export of Log records

### ISSUE THREAD

**Issue by szoupanos** [aiidateam/aiida_core#1759]
The `DbLog` records should be exported and imported properly.
They are important for the new workflow system.

**Comment by szoupanos** [comment]
This is a sample log record generated by a test that is exported:
From a first look, it seems that the objpk is also repeated in the metadata of the record.
It should not be there because it is unnecessary information (repeated) and because the import/export should not be responsible to rewrite & update fields with not predefined schema (JSON)

The objpk will be updated during import to reference the new id of the node that used to reference in the old database.

```
"Log": {
            "2368": {
                "objpk": 2368,
                "loggername": "aiida.aiida.orm.implementation.sqlalchemy.calculation.Calculation",
                "objname": "node.calculation.Calculation.",
                "time": "2018-09-03 14:43:54.055864+02:00",
                "message": "Testing logging of critical failure",
                "levelname": "CRITICAL",
                "metadata": {
                    "objpk": 2368,
                    "threadName": "MainThread",
                    "pathname": "/home/aiida/aiida-code/aiida_core/aiida/backends/tests/export_and_import.py",
                    "name": "aiida.aiida.orm.implementation.sqlalchemy.calculation.Calculation",
                    "module": "export_and_import",
                    "created": 1535978634.055864,
                    "process": 11421,
                    "processName": "MainProcess",
                    "args": [

                    ],
                    "msecs": 55.86409568786621,
                    "filename": "export_and_import.py",
                    "levelno": 50,
                    "thread": 140716069254912,
                    "exc_text": null,
                    "lineno": 1818,
                    "msg": "Testing logging of critical failure",
                    "exc_info": null,
                    "funcName": "test_export_import_of_log_entries",
                    "relativeCreated": 2539.2611026763916,
                    "levelname": "CRITICAL",
                    "objname": "node.calculation.Calculation."
                }
            }
        }
```

**Comment by szoupanos** [comment]
Let me give an update on this and mention the status until today:
- My original implementation was exporting DbLogs and I had implemented a query mechanism for both backend to search log records based on the node ids. This is not needed now since there another mechanism to search for such records implemented at the provenance-redesign branch that uses QueryBuilder
- I changed the DBLog schema (and all the classes depending on this) to support UUIDs. There will not be any objpk but objuuid since when you export records it doesn't make any sense to reference the records with a local pk.
- I have to check that there is no performance degradation when searching the log recors of a node with the uuid instead of the pk (I doubt that there will be an issue but I will check).
- I added a UUID attribute to the log records to reference properly when exported (this seems reasonable but I have to double-check if it really needed).
- I created migrations for the new schema (Django, SQLA missing)
- I have to remove the objuuid from the log metadata (it is not needed since it is already exported as an independent attribute of the record)
- I checked at Antimo's DB and the entities that have a log entry are node entities, so they have a uuid to reference them.

- I will continue with the work on the import mechanism of the DbLog records and the SQLA migration on the 27th of Dec.

**Comment by szoupanos** [comment]
Some overall comments & update:
- both models have been modified to add uuid and objuuid columns. The opjpk column has been removed.
- from now on, nodes are referenced by log records using the uuid
- migrations for both backends have been written
- I kept objuuid in the metadata since I see also other attributes of the log entry there. We should take a general decision of what is added to the metadata (not just remove the objuuid) - I am not sure if it should be done in this ticket
- I did perf tests between the old log schema and the new one and using the objuuid to lookup for nodes is 50% slower than using the pk (both have indexes on the UUID and the pk of the nodes)
- My general comment on the performance degradation after chatting a bit with @sphuber is that such lookups are not common, not critically important (happen only on reports - launched manually by a verdi command) and therefore we both tend to agree that we should keep the import/export code as simple as possible (it is already complicated enough) and not add another step that converts the objpk to opjuuid at that module and keep the old log db schema.

I will do some final changes and I will proceed with a pull request

**Comment by szoupanos** [comment]
Yahoooo!!!

**Comment by szoupanos** [comment]
More changes will happen on this:
- the migrations need to be updated and corrected
  - obj_name is not needed anymore
  - obj_pk will be used instead of obj_uuid and it will be renamed to dbnode_id
  - logs not related to nodes will be exported in JSON files
- DbLog.objname will be removed from the models and the code in general since only node related records will be stored

**Comment by sphuber** [comment]
Fixed in PR #2447
